compose: Fix bolding/italicizing after triple-clicking in Chrome.

anurastogiji opened the pull request:
This PR is a bug fix related to the issue mentioned in #19191 

I have reproduced the following bug according to the issue 
![(23) test here - Zulip Community - Zulip](https://user-images.githubusercontent.com/75037620/131415693-9c4da29c-0603-4aca-84d0-9269c84a32d2.gif)

I have fixed the bug in following way (running on localhost)
![(15) social - Zulip Dev - Zulip](https://user-images.githubusercontent.com/75037620/131415855-018b8cd8-5ada-4ec4-9e3a-c051498c6679.gif)

I looked over the problems with previous PR and solved in this PR.

### Process of solving the issue
- Got to know the bug, and tried to reproduce it.
- Claimed the issue
- Setup the development environment to understand the code base
- Found out the the file in static folder where issue was to be fixed, the file was ```/static/js/compose_ui.js```
- Issue was with the triple click on a particular line in compose area in chrome, it was adding a `new-line character '\n'` at the end of the selected text when <kbd>ctrl</kbd>+<kbd>B</kbd> or <kbd>ctrl</kbd>+<kbd>I</kbd> was pressed.
- I sliced `\n` and appended it to suffix i.e. `**` or `*`

Please suggest the changes required in this PR.

timabbott commented:
@anurastogiji thanks for working on this!  I'm excited to see this progress.

Is it convenient to add tests for this logic in the `compose_ui.js` node test suite?  That'd be a huge help in verifying the logic here and making it maintainable into the future.

timabbott commented:
I'll also mention that https://github.com/zulip/zulip/pull/18922 is what this is merge conflicting with, and we intend to merge a few more chunks of that PR in the coming week or so.

anurastogiji commented:
@timabbott Thank you for reviewing the pull request. 

I've removed the merge conflict and I'm currently working on node tests. I've checked the code and removed the mutation which was caused by earlier pull request.

timabbott commented:
@anurastogiji can you fix the merge conflict using a rebase?  We don't do merge commits in this project.

Check out our [GitHub guide](https://zulip.readthedocs.io/en/latest/git-guide.html) and [commit guidelines](https://zulip.readthedocs.io/en/latest/contributing/version-control.html) for more details.

anurastogiji commented:
@timabbott this issue can not be reproduced!

Thank you!

timabbott commented:
Indeed; I suspect this means Chrome fixed their behavior to match Firefox on this issue.
